### PR TITLE
Add Drivetrain class

### PR DESCRIPTION
**Pull request by mattfbacon**

### CODE REVIEW

**Comment by seadeef** [comment]
Can we implement cubic scaling like we did in [https://github.com/V5Reveng/armbot](https://github.com/V5Reveng/armbot)?

**Comment by seadeef** [comment]
Also there should be a constant for velocity dampening. i.e `velocity *= dampening` where `dampening` < 1

File: src/Robot.cpp [hunk 1]
```diff
@@ -1,6 +1,15 @@
+#include <pros/misc.h>
+
 #include "Robot.hpp"
 
-Robot::Robot(Sensor sensor, Flywheel flywheel) : m_sensor(sensor), m_flywheel(flywheel) {}
+Robot::Robot(Drivetrain drivetrain, Sensor sensor, Flywheel flywheel) : m_drivetrain(drivetrain), m_sensor(sensor), m_flywheel(flywheel) {}
+
+void Robot::update_drivetrain() {
+	m_drivetrain.update(m_controller.get_analog(pros::E_CONTROLLER_ANALOG_LEFT_Y), m_controller.get_analog(pros::E_CONTROLLER_ANALOG_RIGHT_X));
+	if (m_controller.get_digital_new_press(pros::E_CONTROLLER_DIGITAL_B)) {
```
Comment (seadeef) [inline src/Robot.cpp:9]: There shouldn't be calls to both `m_controller.get_digital_new_press` and `m_controller.get_digital`. Replace one with another if there is no specific reason.

File: include/Drivetrain.hpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cstdint>
+#include <pros/motors.hpp>
+
+enum class DrivetrainReferenceFrame {
```
Comment (seadeef) [inline include/Drivetrain.hpp:6]: Probably implement this with regard to Flywheel on-off states.

File: src/Drivetrain.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	}
+
+	m_left_back_motor.move_velocity(forward_backward_axis + left_right_axis);
+	m_left_front_motor.move_velocity(forward_backward_axis + left_right_axis);
+
+	m_right_back_motor.move_velocity(forward_backward_axis - left_right_axis);
+	m_right_front_motor.move_velocity(forward_backward_axis - left_right_axis);
+}
+
+void Drivetrain::next_reference_frame() {
```
Comment (seadeef) [inline src/Drivetrain.cpp:43]: Also implement this in regard to Flywheel on-off states

File: src/Robot.cpp (hunk 1 above)
Comment (mattfbacon) [reply src/Robot.cpp:9]: I didn't use get_digital...

Comment (seadeef) [reply src/Robot.cpp:9]: idc as long as it's consistent. You introduced a new method into the code

Comment (seadeef) [reply src/Robot.cpp:9]: Wait I guess the `get_digital_new_press` should be used everywhere there's a toggle, I just read the docs.

Comment (seadeef) [reply src/Robot.cpp:9]: Just change it all to `get_digital_new_press` please

**Comment by seadeef** [comment]
Approved

**Review by seadeef — APPROVED** [review]
ig I have to submit a review to merge

**Pull request merged**